### PR TITLE
chore: bump deps, add repeated <-> single field example

### PR DESCRIPTION
**Pull request by DmitriyMV**
Bump deps and also show that wire representation of single string is acceptable by repeated and vice versa.

### CODE REVIEW

File: example_test.go [hunk 1]
```diff
@@ -50,3 +50,47 @@ func Example_test3() {
...
+	err = protoenc.Unmarshal(buf, &withString)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("withString = %+v\n", withString)
+
+	// Output:
+	// withRepeated = {First:9 Slc:[a] Other:e}
+	// withString = {First:9 Str:d Last:e}
```
Comment (smira) [inline example_test.go:95]: so it unmarshals _last_ string? but this is also our implementation, not the official one?

Comment (DmitriyMV) [reply example_test.go:95]: No, we are spec compliant here. https://protobuf.dev/programming-guides/encoding/#last-one-wins

Comment (smira) [reply example_test.go:95]: got it, makes sense.

**Comment by smira** [comment]
I guess we should rekres to use GHA as well

**Pull request closed without merging**